### DISCUSSION TITLE
Bug with Livewire V3 navigate.

### DISCUSSION THREAD

**Discussion by WyattCast44** [livewire/livewire#5829] [7. Bugs (archived)] [answered]
Primary source is the AlpineJS repo, reproducing here for visibility: https://github.com/alpinejs/alpine/discussions/3671#

https://github.com/alpinejs/alpine/blob/a9206161f0bf878d41a001197553613cf31cbde9/packages/navigate/src/page.js#L84

## Summary:

The `cloneScriptTag` function is causing a `ReferenceError` due to the variable `attr` not being defined in the `for...of` loop. This bug prevents the function from executing properly when trying to clone script tags, leading to potential issues when dealing with scripts in the HTML document.

## Steps to Reproduce:

Use the `@livewireScriptConfig` functionality
Navigate to the `cloneScriptTag` function in the repository.
Attempt to execute the function with a sample script element containing attributes.

### Expected Result:

The `cloneScriptTag` function should execute successfully and return a cloned script element, replicating the attributes and content of the input script element.

### Actual Result:
The function throws a `ReferenceError`, stating that `attr` is not defined. Consequently, the function fails to perform the cloning operation correctly, leading to unexpected behavior when dealing with script tags in the HTML document.

## Fix:
In the `cloneScriptTag` function, the variable `attr` used in the `for...of` loop needs to be explicitly defined using `const` or `let` to avoid the `ReferenceError`. The corrected code is as follows:

```js
function cloneScriptTag(el) {
    let script = document.createElement('script');

    script.textContent = el.textContent;
    script.async = el.async;

    for (let attr of el.attributes) {
        script.setAttribute(attr.name, attr.value);
    }

    return script;
}
```

## Additional Information:

I discovered this bug while trying the Livewire V3 beta, very possible it is due to something in my code - just thought I would create an issue if anyone else is experiencing this as well. 

Please let me know if you need any further information or assistance. Congratulations on launching V3! 🎉

## Environment:

Operating System: Windows 10
Browser: Google Chrome 114.0.5735.199 

Attachments:

![Screenshot (258)](https://github.com/alpinejs/alpine/assets/17957937/e7c127c1-ca8f-4ff4-839c-0f3bb9cc8d0b)
![Screenshot (259)](https://github.com/alpinejs/alpine/assets/17957937/bf532d55-1a9f-40e3-92fd-563e19f35e80)

**Comment by joshhanley** [comment]
@WyattCast44 thanks for reporting! Let's keep this going on #5842. See my comment there for more details https://github.com/livewire/livewire/discussions/5842?converting=1#discussioncomment-6513899

**Comment by joshhanley** [comment, marked as answer]
This has been fixed in https://github.com/alpinejs/alpine/pull/3685 and should be in the next beta release.
